content/src/service/history/HistoryManagerImpl.ts

```diff
@@ -102,10 +91,16 @@ export class HistoryManagerImpl implements HistoryManager {
         }
     }
 
-    private async getImmutableHistory(): Promise<DeploymentHistory> {
-        const immutableHistory = await this.storage.getImmutableHistory()
-        // Sort from newest to oldest
-        return immutableHistory.reverse()
+
+    /** Before loading the immutable history, we check if it is in fact necessary */
+    private async retrieveNecessaryHistory(from: Timestamp | undefined): Promise<DeploymentHistory> {
+        if (!from || from <= this.immutableTime) {
```

Comment:
I like it!
I think we can improve it, even more, to avoid reading the immutable history when history is called without parameters.